front/app/containers/Authentication/steps/_components/SSOVerificationButton/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import { colors, IconNames } from '@citizenlab/cl2-component-library';
+import { MessageDescriptor } from 'react-intl';
+
+import {
+  TVerificationMethod,
+  TVerificationMethodName,
+} from 'api/verification_methods/types';
+
+import { AUTH_PATH } from 'containers/App/constants';
+
+import VerificationMethodButton from 'components/UI/VerificationMethodButton';
+
+import { getJwt } from 'utils/auth/jwt';
+import { FormattedMessage } from 'utils/cl-intl';
+import { removeUrlLocale } from 'utils/removeUrlLocale';
+
+import messages from './messages';
+
+interface Props {
+  last: boolean;
+  verificationMethodName?: TVerificationMethodName; // TODO: JS - tidy up the fact we have both here
+  verificationMethod?: TVerificationMethod;
+  grayBorder?: boolean;
+  standardSSOBehavior?: boolean;
+  onClick?: (method: TVerificationMethod) => void;
+  onClickStandardSSO?: () => void;
+}
+
+const SSOVerificationButton = ({
+  verificationMethodName,
+  verificationMethod,
+  last,
+  grayBorder,
+  standardSSOBehavior,
+  onClick,
+  onClickStandardSSO,
+}: Props) => {
+  const methodName =
+    verificationMethodName || verificationMethod?.attributes.name;
+  if (!methodName) return null;
+
+  const handleOnClick = () => {
+    if (verificationMethod && onClick) {
+      onClick(verificationMethod);
+    }
+
+    if (standardSSOBehavior) {
+      onClickStandardSSO?.();
+    } else {
+      const jwt = getJwt();
+      window.location.href = `${AUTH_PATH}/${
+        verificationMethod?.attributes.name
+      }?token=${jwt}&pathname=${removeUrlLocale(window.location.pathname)}`;
+    }
```

Comment:
Minor: I wonder if we could do early returns in here. At the moment, I don't think we will have two blocks running after the other but as the code changes, that could happen and cause some issues.

```suggestion
    if (verificationMethod && onClick) {
      onClick(verificationMethod);
      return;
    }

    if (standardSSOBehavior) {
      onClickStandardSSO?.();
      return;
    }
    const jwt = getJwt();
    window.location.href = `${AUTH_PATH}/${
      verificationMethod?.attributes.name
    }?token=${jwt}&pathname=${removeUrlLocale(window.location.pathname)}`;
```

Reply:
Tried this and alas adding return in there breaks the button so it does nothing.